drivers/mma8x5x/mma8x5x.c

```diff
@@ -51,7 +63,7 @@ int mma8x5x_init(mma8x5x_t *dev, const mma8x5x_params_t *params)
 
     /* test if the target device responds */
     i2c_read_reg(BUS, ADDR, MMA8X5X_WHO_AM_I, &reg, 0);
-    if (reg != dev->params.type) {
+    if (reg != MMA8X5X_TYPE) {
```

Comment:
Instead if doint this comparison, and having pseudomodules to change the value of MMA8X5X_TYPE, would it be simple to check if `reg` is equal to any of `MMA8X5X_TYPE_MMA8???`. Especially since, in the rest of the module, there is no other code dependent on the device subtype.

Reply:
What's the point of hard-coding the variant in the module's code? How does it improve anything? It is not possible any longer to have two variants in the same firmware.

Also, there's no code to check that conflicting pseudomodules are not enabled.

I'd argue that checking that the type matches a specific sub-type is close to useless. Even in the case where one has different subtypes connected and wants to differentiate them, it can be done by adding a method that returns the subtype.

Reply:
I think you are commenting on the outdated version. The new version allows both:
- if the firmware should be used with one (or several) sensors of the same type, then the application makefile can just pass the explicit variant using the corresponding pseudo-module. The driver will use the check for that specific version
- if the firmware should be used with multiple and different variant of the sensor, then no pseudo-module is used (just import the generic driver version) and a loose comparison is used (again 0x0a which is common to all variants).

Reply:
I pushed a new version that keep the type parameter but its default value is now defined based on the pseudomodule passed to the build system. Let me know what you think of it.